Challenge 2.0

DOMINION-JOHN1 opened the pull request:
started level 2 challenge. You can at occasion check out my work

DOMINION-JOHN1 committed:
a def function to calculate numbers that can form a triangle
rename function

DOMINION-JOHN1 committed:
create a def function to return multiples of a number in an array

greatertomi commented on CHALLENGE 2.0/def,multiple_3_calc.py:
```diff
@@ -0,0 +1,8 @@
+def mul_calc(num):
+    num=float(input("enter value here; "))
```
Don't accept from this function, only use the `num` parameter.

DOMINION-JOHN1 replied:
resolved

greatertomi commented on CHALLENGE 2.0/def,multiple_3_calc.py:
```diff
@@ -0,0 +1,8 @@
+def mul_calc(num):
+    num=float(input("enter value here; "))
+    arrayA=range(3,100,5)
```
You can safely remove this line and do something like this `for i in range(3,100,5)`

DOMINION-JOHN1 replied:
resolved.

greatertomi commented on CHALLENGE 2.0/def,trian_calc.py:
```diff
@@ -0,0 +1,14 @@
+def trian_calc (A,B,C):
+    A=float(input("insert a number here; "))
+    B=float(input("insert a number here; "))
+    C=float(input("insert a number here; "))
```
Only use the functions paramaters. Remove these

DOMINION-JOHN1 replied:
alright i already removed the input function

greatertomi replied:
Not resolved

greatertomi commented on CHALLENGE 2.0/def,multiple_3_calc.py:
```diff
@@ -0,0 +1,8 @@
+    arrayA=range(3,100,5)
+    for i in arrayA:
+        print(i%num==0)
+
```
Your function does not return anything. I am expecting it to return the answer. Use `return` instead of `print`

DOMINION-JOHN1 replied:
Now resolved

greatertomi commented on CHALLENGE 2.0/def,multiple_3_calc.py:
```diff
@@ -0,0 +1,8 @@
+def mul_calc(num):
+    num=float(input("enter value here; "))
+    arrayA=range(3,100,5)
```
Are you sure `range(3,100,5)` would solve any array of numbers? Because I don't why you are using fixed numbers here.

DOMINION-JOHN1 replied:
yes , i have changed *arg  in the def function so it can collect  an any array of numbers

greatertomi commented on CHALLENGE 2.0/def,trian_calc.py:
```diff
@@ -0,0 +1,14 @@
+    if  A+B > C and  A+C > B and B+C > A:
+        print(True)
+    else:
+        print(False)
```
This function does not return anything. I am expecting it to return the answer. So don't use `print`, return the answer

DOMINION-JOHN1 replied:
done!

greatertomi replied:
You have not resolved this. Use this kind of return

```suggestion
        return False
```

DOMINION-JOHN1 committed:
initial commit

DOMINION-JOHN1 committed:
rename multi_calc def function

DOMINION-JOHN1 committed:
set the def function to receive multiple number in the array

DOMINION-JOHN1 committed:
initial commit

DOMINION-JOHN1 committed:
initial commit

DOMINION-JOHN1 committed:
optimize the application with some new features

DOMINION-JOHN1 committed:
initial commit

DOMINION-JOHN1 commented:
apologies for the delay, i had to finish some modules to be able to solve these challenges.  Do check out what i have being working on. should be done in a day or two.

DOMINION-JOHN1 committed:
add function that display various comments after each answer.

DOMINION-JOHN1 committed:
add function that display various comments after each answer and new files for each application upgrade

DOMINION-JOHN1 committed:
trying a lot of options

DOMINION-JOHN1 commented:
for the part B of this challenge I created a new python file for each modification of the app. example: multi_tutor 1 means the first questions of part b

DOMINION-JOHN1 committed:
trying a lot of options

DOMINION-JOHN1 committed:
add break to end the loop when the coumt reaches 10

DOMINION-JOHN1 committed:
initial commit

DOMINION-JOHN1 commented on CHALLENGE 2.0/def,multiple_3_calc.py:
```diff
@@ -0,0 +1,8 @@
+    num=float(input("enter value here; "))
+    arrayA=range(3,100,5)
+    for i in arrayA:
+        print(i%num==0)
```
oh this has being corrected

DOMINION-JOHN1 replied:
there was suposed to be an If statement inbetween

DOMINION-JOHN1 committed:
rename file and def function.
remove input functions and replace 'print' with return  function

DOMINION-JOHN1 committed:
rename file and def function.
create an unlimited array of numbers using the input function and split method.
remove num and array from the def parameters

DOMINION-JOHN1 committed:
rename file and def function.

DOMINION-JOHN1 committed:
rename file and def function.

DOMINION-JOHN1 committed:
rename file and def function.
use random.choice instead of creating a dictionary to choose a from a list of comment option
add a break function at the end to stop the function if the student fails

DOMINION-JOHN1 committed:
rename file and def function.
add a break function to restart the game when the game count reaches 10
use a counting method to trail when the students attempts reaches 10.
add a break function to stop and restart the game immediately the studebts attempt reaches 10

DOMINION-JOHN1 committed:
rename file and def function.
change the random randint input parameters so it can feature 2 and 3 digits numbers

DOMINION-JOHN1 committed:
rename file and def function.
create a new function to generate questions for the
different problem type and difficulty

greatertomi commented on CHALLENGE 2.0/def,trian_calc.py:
```diff
@@ -0,0 +1,14 @@
+def trian_calc ():
+    A=float(input("insert a number here; "))
+    B=float(input("insert a number here; "))
+    C=float(input("insert a number here; "))
```
You've not removed the input function. Do something like this and use the input function outside this triangle function
```suggestion
def triangle_calculation(side1, side2, side3):
   if  side1+side2 > side3 and  side1+side3 > side2 and side2+side3 > side1
   ...
   ```

greatertomi commented on CHALLENGE 2.0/distance_calculator.py:
```diff
@@ -0,0 +1,9 @@
+def calculate_distance():
+    x1=float(input("enter value here; "))
+    y1=float(input("enter value here; "))
+    x2 = float(input("enter value here; "))
+    y2=float(input("enter value here; "))
```
Don't use input inside this function. Do something like 

```suggestion
def calculate_distance(x1, y1, x2, y2, d):
    ...
```

DOMINION-JOHN1 committed:
rename of variables and functions
remove input functions

DOMINION-JOHN1 commented:
Changes have being pushed to branch challenge _2.0

DOMINION-JOHN1 committed:
Remove break function so the program can repeat the same question until user gets the right answer

DOMINION-JOHN1 committed:
conversion of the answer to strings so it can allow empty and strings input of the user

DOMINION-JOHN1 committed:
conversion of the answer to strings so it can allow empty and strings input of the user

DOMINION-JOHN1 committed:
conversion of the answer to strings so it can allow empty and strings input of the user

DOMINION-JOHN1 committed:
conversion of the answer to strings so it can allow empty and strings input of the user
removal of break function

DOMINION-JOHN1 committed:
conversion of the answer to strings so it can allow empty and strings input of the user
removal of break function

DOMINION-JOHN1 committed:
delete def,trian_calc

DOMINION-JOHN1 committed:
Delete multi_tutor_3.py

DOMINION-JOHN1 committed:
Delete multi calc

DOMINION-JOHN1 committed:
Delete multi_tutor_1.py

greatertomi commented on CHALLENGE 2.0/distance_calculator.py:
```diff
@@ -0,0 +1,5 @@
+def calculate_distance(x1,y1,x2,y2):
+    distance=(((x2-x1)**2) +((y2-y1)**2))**0.5
+    print(distance)
```
Use `return` instead of `print`

greatertomi commented on CHALLENGE 2.0/distance_calculator.py:
```diff
@@ -0,0 +1,5 @@
+def calculate_distance(x1,y1,x2,y2):
+    distance=(((x2-x1)**2) +((y2-y1)**2))**0.5
+    print(distance)
+## call the function and input the desired variables
```
Comment should be one "#"

greatertomi commented on CHALLENGE 2.0/distance_calculator.py:
```diff
@@ -0,0 +1,5 @@
+    distance=(((x2-x1)**2) +((y2-y1)**2))**0.5
+    print(distance)
+## call the function and input the desired variables
+calculate_distance(4,5,6,7)
```
You have to capture edge cases, don't just test the happy path scenario. Make sure these returns a proper answer

```
calculate_distance(4,5) -> throw an error that some values are missing
calculate_distance(4,5, 'b', '2') -> throw an error that you only accept integers
calculate_distance(4,5, '6', '2.56') -> convert the strings to number and calculate with them
```

DOMINION-JOHN1 replied:
done with the third
i am not too clear on this
for the first two you want me to add them as an error message?

greatertomi commented on CHALLENGE 2.0/find_multiples_in_array.py:
```diff
@@ -0,0 +1,8 @@
+def get_multiples(arrayA,num):
+    arrayA = [2, 4, 6, 8, 10, 12, 14]
+    num = 2
```
Pass the `arrayA` and `num` from outside the function, because here you are not using the parameters.

greatertomi commented on CHALLENGE 2.0/find_multiples_in_array.py:
```diff
@@ -0,0 +1,8 @@
+def get_multiples(arrayA,num):
+    arrayA = [2, 4, 6, 8, 10, 12, 14]
+    num = 2
+    return [x for x in arrayA if x % num == 0]
```
I like this list comprehension. Smart stuff

DOMINION-JOHN1 replied:
Finally! my head is swelling. Thanks

greatertomi commented on CHALLENGE 2.0/find_multiples_in_array.py:
```diff
@@ -0,0 +1,8 @@
+    arrayA = [2, 4, 6, 8, 10, 12, 14]
+    num = 2
+    return [x for x in arrayA if x % num == 0]
+result = get_multiples('arrayA','num')
```
Passing strings that you didn't use into the function doesn't make sense.

DOMINION-JOHN1 replied:
done

greatertomi commented on CHALLENGE 2.0/unique_string.py:
```diff
@@ -0,0 +1,4 @@
+def remove_duplicates(*args):
+    print(set(args))
```
use `return` instead of `print`

DOMINION-JOHN1 replied:
true! corrected .

greatertomi commented on CHALLENGE 2.0/unique_string.py:
```diff
@@ -0,0 +1,4 @@
+def remove_duplicates(*args):
```
This function should recieve a string, not *args. This function can't pass the testcases in the question

```
remove_duplicates('AABBBCCCDEEEE')
remove_duplicates('GHIIIKKL')
```

greatertomi commented on CHALLENGE 2.0/numbers_that_form_triangle.py:
```diff
@@ -0,0 +1,11 @@
+def triangle_num(a, b, c):
```
You always assume your function would receive a happy-path parameter. Could you validate the inputs here before doing the calculation?

DOMINION-JOHN1 replied:
i dont grab this. By validate do you mean i  should make the function accept other data type?

greatertomi commented on CHALLENGE 2.0/numbers_that_form_triangle.py:
```diff
@@ -0,0 +1,11 @@
+def triangle_num(a, b, c):
+    if (a + b > c) and (b + c > a) and (c + a > b):
```
It would also be good to improve on the variable naming instead of a, b and c

DOMINION-JOHN1 committed:
resolve of reviews from pull request

DOMINION-JOHN1 committed:
Merge remote-tracking branch 'origin/challenge_2.0' into challenge_2.0

DOMINION-JOHN1 committed:
addition of codes to handle error and validate inputs of the functions

greatertomi commented on CHALLENGE 2.0/unique_string.py:
```diff
@@ -0,0 +1,5 @@
+    return set(string_list)
+
+
+print(remove_duplicates('namedominionjohneleojo'))
```
This returns `{'m', 'h', 'a', 'i', 'l', 'o', 'j', 'e', 'd', 'n'}`. It should return a string like "adelhijmno"

greatertomi commented on CHALLENGE 2.0/guess_the_number_calc.py:
Why do I always get the answer after every second try? The generated number was 368, for example. I entered 500 first try, then 250 second try and the app said I was correct.

Check the logic of this

greatertomi commented on CHALLENGE 2.0/maths_multiplication_tutor_legend.py:
- This has all the above errors I wrote about level2 and level3. 
- In division, the app does not mark my right answer and maybe you should approximate division answers to 2 decimal places (don't forget to notify the users). 
- Selecting mixture questions only brings up division problems

greatertomi commented on CHALLENGE 2.0/maths_multiplication_tutor_level2.py:
Correct answer always throws "Excellent!" and the wrong answer always throws “No. Keep trying“. The responses are not varied.

greatertomi commented on CHALLENGE 2.0/multiplication_tutor_level3.py:
- ~Why does it allow me to reanswer questions I failed?~
- It doesn’t grade. I can answer more than 10 questions. It should restrict to only 10 questions

I feel you don’t even check the requirements of the question. After coding, check if your code meets the requirements.

DOMINION-JOHN1 committed:
addition of codes to handle error and validate inputs of the functions
addition of code for a replay if player picks

DOMINION-JOHN1 committed:
addition of codes for different responses after each wrong or right answer

DOMINION-JOHN1 committed:
addition of codes for different responses after each wrong or right answer and to automatically stop after 10 tries.

DOMINION-JOHN1 committed:
include range of numbers that are 3 digits

DOMINION-JOHN1 committed:
add code to put answers on the diviion data type to 2 decimal place

DOMINION-JOHN1 committed:
addition of code to make calc generate new questions after a failed attempt

DOMINION-JOHN1 committed:
addition of code to add a problem type that randomly mixes arithemetic options

DOMINION-JOHN1 committed:
addition of code to add a problem type that randomly mixes arithemetic options .
arrangement of codes

DOMINION-JOHN1 committed:
Delete multplication_tutor_master.py

DOMINION-JOHN1 committed:
Update maths_multiplication_tutor_level5.py

DOMINION-JOHN1 committed:
Update multiplication_level_python_tutor_3

DOMINION-JOHN1 committed:
Update multplication_tutor_level4.py

DOMINION-JOHN1 committed:
Update unique_string.py

DOMINION-JOHN1 commented:
Thanks sir for the approval
